### PR TITLE
[SGTM Deployments] Download linux-compatible deps for lambda bundle; switch virtual environment management to pipenv

### PR DESCRIPTION
**Pull request by harshita-gupta**
### Summary

@vn6 was running into `ERROR] Runtime.ImportModuleError: Unable to import module 'src.handler': /var/task/cryptography/hazmat/bindings/_rust.abi3.so: invalid ELF header` while trying to deploy SGTM from #169. This is an issue with the wrong cryptography binary being run on the lambda - we need to install the linux wheel for cryptography. I'm somewhat surprised that this hasn't broken before; it could be that the new botocore/boto3 code we added is finally leading to us hitting an implementation difference across wheels.

As part of this change, I also migrated python environment management to pipenv for sgtm. This'll make our builds a litle bit more standard across SGTM deployers.

Asana tasks:
https://app.asana.com/0/1200740774079340/1207164718244879/f

Relevant deployment: 

CC: @vn6 @suzyng83209 @prebeta

### Risks

Pull Request: https://github.com/Asana/SGTM/pull/170

Pull Request synchronized with [Asana task](https://app.asana.com/0/0/1207165201769832)

### CODE REVIEW

**Comment by harshita-gupta** [comment]
* **#170** <a href="https://app.graphite.dev/github/pr/Asana/SGTM/170?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @harshita-gupta and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: test/asana/helpers/test_asana_comment_from_github_comment.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from html import escape
+import os
```
Comment (suzyng83209) [inline test/asana/helpers/test_asana_comment_from_github_comment.py:2]: we need os here?

**Review by suzyng83209 — APPROVED** [review]
1 nit on import but lgtm

File: test/asana/helpers/test_asana_comment_from_github_comment.py (hunk 1 above)
Comment (vn6) [reply test/asana/helpers/test_asana_comment_from_github_comment.py]: ah we don't, removed!

**Pull request merged**